api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -1650,4 +1650,17 @@ private void withCurrentUserAs(User user, Runnable internals) throws IllegalAcce
 			FieldUtils.getField(UserContext.class, "user", true).set(userContext, authenticatedUser);
 		}
 	}
+
+	@Test
+	public void saveUserProperty_shouldAddANewPropertyWithAVeryLargeStringWithoutRunningIntoError() {
+		final String USER_PROPERTY_KEY = liquibase.util.StringUtil.repeat("emrapi.lastViewedPatientIds,",10);
+		final String USER_PROPERTY_VALUE = liquibase.util.StringUtil.repeat("1234",500);
+		User updatedUser = userService.saveUserProperty(USER_PROPERTY_KEY, USER_PROPERTY_VALUE);
+		String keyList = "";
+		for (String key : updatedUser.getUserProperties().keySet()) {
```

Comment:
What is the use of this loop?

Reply:
To iterate over  updatedUser.getUserProperties().keySet() so i can pick the keys . I could not pick the key from  updatedUser.getUserProperties()